test/UnorderedNonce.t.sol

```diff
@@ -0,0 +1,113 @@
+        unorderedNonce.spendNonce(address(this), 255);
+        vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+        unorderedNonce.spendNonce(address(this), 256);
+    }
+
+    function testHighNonces() public {
+        unorderedNonce.spendNonce(address(this), 2 ** 240);
+        unorderedNonce.spendNonce(address(this), 2 ** 240 + 1);
+
+        vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+        unorderedNonce.spendNonce(address(this), 2 ** 240);
+        vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+        unorderedNonce.spendNonce(address(this), 2 ** 240 + 1);
+
+        unorderedNonce.spendNonce(address(this), 2 ** 240 + 2);
+    }
+
+    function testInvalidateFullWord() public {
+        unorderedNonce.invalidateUnorderedNonces(0, 2 ** 256 - 1);
+
+        vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+        unorderedNonce.spendNonce(address(this), 0);
+        vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+        unorderedNonce.spendNonce(address(this), 1);
+        vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+        unorderedNonce.spendNonce(address(this), 254);
+        vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+        unorderedNonce.spendNonce(address(this), 255);
+        unorderedNonce.spendNonce(address(this), 256);
+    }
+
+    function testInvalidateNonzeroWord() public {
+        unorderedNonce.invalidateUnorderedNonces(1, 2 ** 256 - 1);
+
+        unorderedNonce.spendNonce(address(this), 0);
+        unorderedNonce.spendNonce(address(this), 254);
+        unorderedNonce.spendNonce(address(this), 255);
+        vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+        unorderedNonce.spendNonce(address(this), 256);
+        vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+        unorderedNonce.spendNonce(address(this), 511);
+        unorderedNonce.spendNonce(address(this), 512);
+    }
+
+    function testUsingNonceTwiceFails(uint256 nonce) public {
+        unorderedNonce.spendNonce(address(this), nonce);
+        vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+        unorderedNonce.spendNonce(address(this), nonce);
+    }
+
+    function testUseTwoRandomNonces(uint256 first, uint256 second) public {
+        unorderedNonce.spendNonce(address(this), first);
+        if (first == second) {
+            vm.expectRevert(UnorderedNonce.NonceAlreadyUsed.selector);
+            unorderedNonce.spendNonce(address(this), second);
+        } else {
+            unorderedNonce.spendNonce(address(this), second);
+        }
+    }
+
+    function testInvalidateNoncesRandomly(uint248 wordPos, uint256 mask) public {
+        unorderedNonce.invalidateUnorderedNonces(wordPos, mask);
+        assertEq(mask, unorderedNonce.nonces(address(this), wordPos));
+    }
+
+    function testInvalidateTwoNoncesRandomly(uint248 wordPos, uint256 startBitmap, uint256 mask) public {
```

Comment:
i feel like this test doesnt actually check the unordered nonce contract? it just tests the `invalidateUnorderedNonces` function which doesnt use UnorderedNonce it just implements something custom?

e.g. there can easily be overlap between `startBitmap` and `mask` having the same bit set... and that doesnt revert because its not actually using the nonce checking.

Honestly i just think this test and the one above can be deleted? The test `testInvalidateNonzeroWord` had a nice use of this function to then check that used nonces reverted